tools/c7n_azure/c7n_azure/resources/postgresql_server.py

```diff
@@ -69,3 +71,77 @@ def _query_rules(self, resource):
                 continue
             resource_rules.add(rule)
         return resource_rules
+
+
+@PostgresqlServer.filter_registry.register('configuration-parameters')
+class ConfigurationParametersFilter(ValueFilter):
+    """Filter by configuration parameter for this postresql server
+
+    Configurations are made available to the filter as a map with each
+    key holding the name of the configuration and each value holding
+    the properties of the Configuration as defined here:
+    https://learn.microsoft.com/en-us/python/api/azure-mgmt-rdbms/azure.mgmt.rdbms.postgresql.models.configuration?view=azure-python
+
+    :example:
+
+    Example JSON document showing the data format provided to the filter
+
+    .. code-block:: json
+
+      {
+        log_connections: {
+          "id": "<example-id>",
+          "name": "log_connections",
+          "type": "Microsoft.DBforPostgreSQL/servers/configurations",
+          "value": "off",
+          "description": "Logs each successful connection.",
+          "default_value": "on",
+          "data_type": "Boolean",
+          "allowed_values": "on,off",
+          "source": "user-override"
+        },
+        {
+          "id": "<example-id>",
+          "name": "log_min_duration_statement",
+          "type": "Microsoft.DBforPostgreSQL/servers/configurations",
+          "value": "-1",
+          "description": "example description",
+          "default_value": "-1",
+          "data_type": "Integer",
+          "allowed_values": "-1-2147483647",
+          "source": "system-default"
+        }
+      }
+
+    :example:
+
+    Find Postgresql servers with log_connections not enabled
+
+    .. code-block:: yaml
+
+        policies:
+          - name: sql-database-no-log-connections
+            resource: azure.postgresql-server
+            filters:
+              - type: configuration-parameters
+                key: log_connections.value
+                op: ne
+                value: 'on'
+
+    """
+    schema = type_schema('configuration-parameters', rinherit=ValueFilter.schema)
+
+    def __call__(self, resource):
+        key = 'configurations'
+        if key not in resource['properties']:
+            client = self.manager.get_client()
+            query = client.configurations.list_by_server(
```

Comment:
any reason for list vs get here?

Reply:
The way the filter is currently structured there is no easy way to know exactly which parameter would be evaluated since the `key` field could be any JMESPath expression. But I think it makes sense to refactor the filter a bit so that it accepts a single parameter name to evaluate as that would be the most common use case.